src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage.cs

```diff
@@ -2,39 +2,34 @@
 
 using System;
 using System.Threading;
-using System.Windows.Threading;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
-using Microsoft.CodeAnalysis.Utilities;
+using Microsoft.VisualStudio.ComponentModelHost;
 using Microsoft.VisualStudio.Shell;
 using Task = System.Threading.Tasks.Task;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
 {
     internal abstract class AbstractPackage : AsyncPackage
     {
-        protected ForegroundThreadAffinitizedObject ForegroundObject;
+        protected ForegroundThreadAffinitizedObject ForegroundObject
+        {
+            get;
+            private set;
+        }
 
         protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
             await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
 
             await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            // Assume that we are being initialized on the UI thread at this point, and setup our foreground state
-            var kind = ForegroundThreadDataInfo.CreateDefault(ForegroundThreadDataKind.ForcedByPackageInitialize);
-
-            // None of the work posted to the foregroundTaskScheduler should block pending keyboard/mouse input from the user.
-            // So instead of using the default priority which is above user input, we use Background priority which is 1 level
-            // below user input.
-            var taskScheduler = new SynchronizationContextTaskScheduler(new DispatcherSynchronizationContext(Dispatcher.CurrentDispatcher, DispatcherPriority.Background));
-
-            ForegroundThreadAffinitizedObject.CurrentForegroundThreadData = new ForegroundThreadData(Thread.CurrentThread, taskScheduler, kind);
-            ForegroundObject = new ForegroundThreadAffinitizedObject();
+            var componentModel = (IComponentModel)await GetServiceAsync(typeof(SComponentModel));
+            ForegroundObject = new ForegroundThreadAffinitizedObject(componentModel.GetService<IThreadingContext>());
         }
 
         protected void LoadComponentsInUIContextOnceSolutionFullyLoaded(CancellationToken cancellationToken)
         {
-            ForegroundObject.AssertIsForeground();
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
Why this change? There are other places whre ForegroundObject is still being used for assertions, so change seems odd (and using icky statics...)

Reply:
➡️ This change was unintentionally included at this point. `AbstractPackage.ForegroundObject` will be removed soon. We can either consider this tracked already by the work to remove `ForegroundThreadAffinitizedObject`, or we can create a separate issue for it (added to the list just in case).